lib/spack/spack/ci.py

```diff
@@ -757,6 +942,27 @@ def generate_gitlab_ci_yaml(
     shutil.copyfile(env.manifest_path, os.path.join(concrete_env_dir, "spack.yaml"))
     shutil.copyfile(env.lock_path, os.path.join(concrete_env_dir, "spack.lock"))
 
+    with open(env.manifest_path, "r") as env_fd:
```

Comment:
@alalazo @tgamblin do you guys know of a better way to do this? The goal here is to add the spack config paths to the concrete environment yaml `spack.include` section so the extra config scopes only need to be specified once in the root `.gitlab-ci.yaml` rather than having to add them every time we call spack for CI.

One issue I would like to resolve is that the includes are kind of fragile. One issue that we found was the reproducer puts SPACK_ROOT in a different path than CI. This is fixable in the reproducer, but it would be nice to be able to point those includes at something like `$spack/share/spack/gitlab/cloud_pipelines/...` similarly to how the `config` section handles paths.

Another issue is there is no programmatic or composition method available to add includes to an environment. I could make one by moving includes into a "section" schema but I am not sure if that is be best course since that could in theory create recursive includes where includes include scopes of the includes that included the include scope.